pkg/controller/attach_detach.go

```diff
@@ -759,12 +760,18 @@ func (r *ReconcileAttachDetach) recoverAzVolumeAttachment(ctx context.Context, r
 				if azv.Status.State != targetState {
 					_, err = updateState(azv, targetState, forceUpdate)
 				}
+				// clear detail and error in case of recovery
+				if clearDetail {
+					azv.Status.Detail = nil
+					azv.Status.Error = nil
+				}
 				return err
 			}
 			switch azv.Status.State {
 			case azdiskv1beta2.Attaching:
 				// reset state to Pending so Attach operation can be redone
 				targetState = azdiskv1beta2.AttachmentPending
+				clearDetail = true
```

Comment:
Rather than use a Boolean flag referenced by the update function to determine whether or not to clear the `Status.Detail` field, it would be cleaner to wrap `updateFunc` in another function that clears the detail, e.g.

```go
innerUpdateFunc := updateFunc
updateFunc := func(obj client.Object) error {
    if err := innerUpdateFunc(obj); err != nil {
        return err
    }

    azv := obj.(*azdiskv1beta2.AzVolumeAttachment)
    azv.Status.Detail = nil
    azv.Status.Error = nil
}
```